x-pack/plugins/infra/public/plugin.ts

```diff
@@ -178,10 +178,19 @@ export class Plugin implements InfraClientPluginClass {
       category: DEFAULT_APP_CATEGORIES.observability,
       mount: async (params: AppMountParameters) => {
         // mount callback should not use setup dependencies, get start dependencies instead
-        const [coreStart, pluginsStart, pluginStart] = await core.getStartServices();
-        const { renderApp } = await import('./apps/logs_app');
+        const [coreStart, plugins, pluginStart] = await core.getStartServices();
 
-        return renderApp(coreStart, pluginsStart, pluginStart, params);
+        if (this.config.logs.app_target === 'discover') {
```

Comment:
I wonder if we shouldn't replace the whole app registration, since we don't want the deep links to be registered either. What do you think?

Reply:
What would be different apart from removing the deep links from the other registration? Would this end up registering the app with the same configuration and a different mount app depending on the configuration?

Reply:
Good question 🤔 right now I can think of these:

- the `mount` functions mounts the discover redirect app
- the `navLinkStatus` is set to hidden
- no deep links are registered